src/crewai/crew.py

```diff
@@ -255,12 +256,16 @@ def kickoff(
         i18n = I18N(prompt_file=self.prompt_file)
 
         for agent in self.agents:
+            # type: ignore # Argument 1 to "_interpolate_inputs" of "Crew" has incompatible type "dict[str, Any] | None"; expected "dict[str, Any]"
             agent.i18n = i18n
-            agent.crew = self
-
-            if not agent.function_calling_llm:
+            # type: ignore[attr-defined] # Argument 1 to "_interpolate_inputs" of "Crew" has incompatible type "dict[str, Any] | None"; expected "dict[str, Any]"
+            agent.crew = self  # type: ignore[attr-defined]
+            if (
+                hasattr(agent, "function_calling_llm")
+                and not agent.function_calling_llm
+            ):
                 agent.function_calling_llm = self.function_calling_llm
-            if not agent.step_callback:
+            if hasattr(agent, "step_callback") and not agent.step_callback:
```

Comment:
This may be good-enough for now, but as we want to go down the type-safety road, I would replace it with a check (and cast) for something like `AgentWithCallbacks` protocol.

We could add a TODO comment, so we don't forget about it.

Reply:
yeah adding a TODO here